File: frontend/src/types/strideEvaluations.ts
export type EvaluationStatus = "pending" | "completed" | "failed";
export type Recommendation = "apply" | "monitor" | "skip" | "needs_review";
export type ConfidenceLevel = "low" | "medium" | "high";

export interface EvaluationSection {
  status?: string;
  score?: number | string | null;
  notes?: string | null;
  evidence?: string[];
  matched_keywords?: string[];
  missing_keywords?: string[];
  [key: string]: unknown;
}

export interface EvidenceItem {
  code?: string;
  message?: string;
  evidence?: string | null;
  status?: string;
  dimension?: string;
  severity?: string;
  score?: number;
  keywords?: string[];
}

export interface StrideEvaluation {
  id: string;
  user_id: string;
  role_id: string;
  evaluation_status: EvaluationStatus;
  overall_score: string | null;
  recommendation: Recommendation | string | null;
  confidence_level: ConfidenceLevel | string | null;
  summary: string | null;
  strengths: EvidenceItem[];
  concerns: EvidenceItem[];
  resume_alignment: EvaluationSection;
  compensation_alignment: EvaluationSection;
  seniority_alignment: EvaluationSection;
  remote_alignment: EvaluationSection;
  technical_alignment: EvaluationSection;
  company_risk: EvaluationSection;
  ats_keywords: string[];
  missing_keywords: string[];
  raw_evaluation_json: {
    ai_status?: "completed" | "failed" | "skipped";
    ai_failure_reason?: string;
    ai_error_type?: string;
    ai_result?: {
      unsupported_claim_warnings?: EvidenceItem[];
      evidence_matches?: EvidenceItem[];
      evidence_gaps?: EvidenceItem[];
      positioning_opportunities?: EvidenceItem[];
      [key: string]: unknown;
    };
    active_resume_source?: {
      source_id: string;
      source_name: string;
      source_type: string;
      version_id: string;
      version_label: string;
    } | null;
    [key: string]: unknown;
  };
  created_at: string;
  updated_at: string;
}

export interface StrideEvaluationCreatePayload {
  user_notes?: string | null;
  user_context?: Record<string, unknown>;
}

export type EvaluationSummaryState =
  | { status: "loading" }
  | { status: "not_evaluated" }
  | { status: "error"; message: string }
  | { status: "completed"; evaluation: StrideEvaluation };

File: backend/app/schemas/stride_evaluations.py
import uuid
from datetime import datetime
from decimal import Decimal
from typing import Any

from pydantic import BaseModel, ConfigDict, Field

from app.constants import StrideEvaluationStatus


class StrideEvaluationCreate(BaseModel):
    user_notes: str | None = Field(default=None, max_length=5000)
    user_context: dict[str, Any] = Field(default_factory=dict)


class StrideEvaluationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID
    role_id: uuid.UUID
    evaluation_status: StrideEvaluationStatus
    overall_score: Decimal | None
    recommendation: str | None
    confidence_level: str | None
    summary: str | None
    strengths: list[dict[str, Any]]
    concerns: list[dict[str, Any]]
    resume_alignment: dict[str, Any]
    compensation_alignment: dict[str, Any]
    seniority_alignment: dict[str, Any]
    remote_alignment: dict[str, Any]
    technical_alignment: dict[str, Any]
    company_risk: dict[str, Any]
    ats_keywords: list[str]
    missing_keywords: list[str]
    raw_evaluation_json: dict[str, Any]
    created_at: datetime
    updated_at: datetime
